core/src/main/java/io/github/tcdl/msb/impl/ResponderServerImpl.java

```diff
@@ -47,7 +47,7 @@ static ResponderServerImpl create(String namespace,  MessageTemplate messageTemp
     public ResponderServer listen() {
         ChannelManager channelManager = msbContext.getChannelManager();
 
-        channelManager.subscribe(namespace,
+        boolean isAlreadySubscribed = channelManager.subscribe(namespace,
```

Comment:
Sounds very complicated. Maybe we should reconsider CollectorManager? The root cause is obviously in subscribe method and maybe we should fix it there